checks/all_checks.go

```diff
@@ -16,18 +16,43 @@
 package checks
 
 import (
+	"os"
+
 	"github.com/ossf/scorecard/v4/checker"
 )
 
-// AllChecks is the list of all security checks that will be run.
-var AllChecks = checker.CheckNameToFnMap{}
+// allChecks is the list of all registered security checks
+var allChecks = checker.CheckNameToFnMap{}
+
+func getAll(overrideExperimental bool) checker.CheckNameToFnMap {
+	// need to make a copy or caller could mutate original map
+	possibleChecks := checker.CheckNameToFnMap{}
+	for k, v := range allChecks {
+		possibleChecks[k] = v
+	}
+
+	if overrideExperimental {
+		return possibleChecks
+	}
+
+	// TODO: remove this check when v6 is released
+	if _, v6 := os.LookupEnv("SCORECARD_V6"); !v6 {
+		delete(possibleChecks, CheckWebHooks)
+	}
 
-// GetAll returns the full list of checks, given any environment variable
+	return possibleChecks
+}
+
+// GetAllForEnvironment returns the full list of checks, given any environment variable
 // constraints.
-// TODO(checks): Is this actually necessary given `AllChecks` exists?
+func GetAllForEnvironment() checker.CheckNameToFnMap {
+	return getAll(false /*overrideExperimental*/)
```

Comment:
Nit: change fn names to `GetAllWithExperimental` and `GetAll`. Wdyt?

Reply:
I played around with a few naming ideas, including those two. 
Personally, it slightly felt off to have a function called `GetAll` which doesn't actually return everything.
But that's just how it felt, if this is a normal pattern I'm fine with adopting those names